functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierConfiguration.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.supplier.zeromq;
+
+import java.util.function.Supplier;
+
+import org.zeromq.ZContext;
+import reactor.core.publisher.Flux;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.channel.FluxMessageChannel;
+import org.springframework.integration.zeromq.inbound.ZeroMqMessageProducer;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * A source module that receives data from ZeroMQ.
+ *
+ * @author Daniel Frey
+ */
+@Configuration
+@EnableConfigurationProperties(ZeroMqSupplierProperties.class)
+public class ZeroMqSupplierConfiguration {
+
+	@Autowired
+	private ZeroMqSupplierProperties properties;
+
+	private FluxMessageChannel output = new FluxMessageChannel();
+
+	@Bean
+	public ZContext zContext() {
+		return new ZContext();
+	}
+
+	@Bean
+	public ZeroMqMessageProducer adapter(ZContext zContext) {
```

Comment:
We probably need to have a `Consumer<ZMQ.Socket> socketConfigurer` as optional dependency.
End-user might need some socket options to provide, like security or read timeout options.
I'm not sure that we can expose all the socket options as a configuration properties (see `SocketBase.setSocketOpt(int option, Object optval)`), but would be great to think about some general.

Reply:
How about I expose the `socketConfigurer` as a `@Bean` so that a user could come in and supply their own version if they would need to override it?

Reply:
Do something like:

```
	@Bean
	@ConditionalOnMissingBean
	public Consumer<ZMQ.Socket> socketConfigurer() {
		return (socket) -> { };
	}
```
Then inject it into the adapter bean creation